### DISCUSSION TITLE
Is "Edit commit message" possible when merging?

### DISCUSSION THREAD

**Discussion by echarrod** [renovatebot/renovate#25766] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.58.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

The PR body with all the current information is included in the PR, but when squashing and merging, a more concise version is used, without any markdown. 

e.g. the table, URLs, and markdown are useful in the MR, but when viewed as the commit, it is pretty messy:
![image](https://github.com/renovatebot/renovate/assets/1381991/e4bc8434-edb7-4de3-ad5a-1a178d37d50d)

When I merge a MR manually, I see this option, so was hoping to be able to utilise it in Renovate:
![image](https://github.com/renovatebot/renovate/assets/1381991/43bfef71-be0d-47f5-80b7-8a73ed79e27e)

### What you tried so far.

I don't want to edit the PR or original commit details, as I still would like this. It is just the "final" commit when squashing and merging

**Comment by rarkins** [comment]
I don't see you mention it, but I assume you are referring to when Renovate does its own automerge? And not using platform automerge?

**Reply by echarrod** [reply]
I would hope for it to apply to automerge types, and regular merges. For the latter so that we don't have to manually edit the commit message each time

**Reply by rarkins** [reply]
You didn't answer my questions

**Reply by echarrod** [reply]
When I said:
> I would hope for it to apply to automerge types

I thought I was answering your question, in that I would hope for it to apply to both Renovate automerges and platform automerges. 

We have set
```json
      "automerge": true,
``` 
in config. And since I see platformAutoMerge defaults to true, I guess we're using that

**Reply by rarkins** [reply]
The next step to work out is which API is being called for that automerge, and does that API support a custom commit message?

**Reply by echarrod** [reply]
For GitLab, it looks like "Merge a Merge Request" section on the [docs ](https://docs.gitlab.com/ee/api/merge_requests.html#merge-a-merge-request)has a `merge_commit_message` you can specify

**Reply by rarkins** [reply]
Then we can populate that with the prTitle contents
